transport/grpc/peer.go

```diff
@@ -83,6 +83,16 @@ func (p *grpcPeer) monitorConnectionStatus() {
 		yarpcStatus := grpcStatusToYARPCStatus(grpcStatus)
 		p.setConnectionStatus(yarpcStatus)
 
+		// When active connection falling back to IDLE state, no automatic reconnection is happening.
+		// There two options:
+		// - Either try to make a call using this connection, which will trigger reconnection, but may lead
+		// to a failed request (host is unreachable or context deadline happened).
+		// - Or we may try to reconnect manually, and use connection only when it's established and explicitly available.
+		// We choose second option.
+		if grpcStatus == connectivity.Idle {
+			p.clientConn.Connect()
+		}
```

Comment:
Let's wrap this up in a exponential backoff or simple wait to avoid a connect busy loop in case the destination host is unavailable. We have seen cases where grpc clients are aggressively retry cause server side overload, even a tiny bit of backoff would be nice

Reply:
Fortunately we don't have to implement it manually, as it's embedded into Connect method: [source code](https://github.com/grpc/grpc-go/blob/v1.41.1/clientconn.go#L1168), and it's global backoff, yes (i.e. it counts attempts and timeouts for all Connect attempts, not within one call).

Reply:
This seems to be using a default backoff which has a long with a very long max-delay of 120s, I'd suggest we bring down the maximum backoff delay to something under 15s.

https://github.com/grpc/grpc-go/blob/78ebe1c6cc888f74a1dae496012f82c4f7d8aeb4/backoff/backoff.go#L47-L52

Reply:
Hmm.. What we're doing with upgrade is basically extracting implicit "Connect" call from inside grpc library to explicit call from yarpc. It means that backoff is already like this. So I think we should adjust only if we have a reason to do so (and in a dedicated PR).